Add __VA_OPT__(,) on print macro

TiagoTeixeira-synthara opened the pull request:
Hey mate, was trying to use this project and encountered this problem:

If the `__VA_ARGS__` is empty, the _trailing_ comma will be a problem. The `__VA_OPT__(,)` should solve this.

I'm creating as a draft in case I find some other suggestions while I'm using this.

Cheers

TiagoTeixeira-synthara committed:
Add __VA_OPT__(,) on print macro

TiagoTeixeira-synthara committed:
Allow parameterized tests

This may be useful to reuse code (and possibly for automatic fuzzing?)

TiagoTeixeira-synthara commented:
I also added the possibility to have parameters in tests, which I find useful to deduplicate code (all those similar edge cases...)

rwypior commented:
Hi,

Those are some nice changes, but I'd like to consider placing those changes in separate macros, like UNITTEST_EXEC_P so we won't break backwards-compatibility (for example in some older microcontroller applications that I still need to support).

Thanks!

TiagoTeixeira-synthara commented:
I believe that as long as the preproccessor support the `__VA_OPT__` macro it should work, anyway I'll make the change as it indeed makes sense

TiagoTeixeira-synthara committed:
Extract parameterized macros to their own

As suggested, parameterized unit tests macros are extracted to their own `_P` macros

TiagoTeixeira-synthara committed:
forgot something in previous commit
